File: script-gen-ui/src/app/model/ipcData.ts
import { IIpcDataInterface } from './interface';

export class IpcData {
  request_type: string;
  additional_info: string;
  json_value: string;
  request_id?: string;

  constructor(data: IIpcDataInterface) {
    this.request_type = data.request_type;
    this.additional_info = data.additional_info;
    this.json_value = data.json_value;
    this.request_id = data.request_id;
  }
}

File: kic-script-gen/src/back_end/ipc_data.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct IpcData {
    pub request_type: String,
    pub additional_info: String,
    pub json_value: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub request_id: Option<String>,
}
